### DISCUSSION TITLE
TemplateBinding don't work in TextBox.InnerLeftContent?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#12893] [General]
Hi, I don't understand why TemplateBinding don't work in the following case:

I have a TemplatedControl containing a TextBox in the template, and I would like to bind an Icon into the InnerLeftContent of the textbox

However, I don't understand why my icon is not showing with this code:
```
  <TextBox.InnerLeftContent>
                                <ContentPresenter Name="PART_IconPresenter" Content="{TemplateBinding Icon}" />
                        </TextBox.InnerLeftContent>
```
if I move the ContentPresenter outside of InnerLeftContent, it works
if I replace ContentPresenter with a static icon with no TemplateBinding, it works too

There is something I don't understand here... anyone have an idea? I see InnerLeftContent, as any other ContentControl, but not...

**Comment by timunie** [comment]
knwon issue #11912

**Reply by dbriard** [reply]
Ah, thank you! I managed to get it work using Binding with ElementName as temporary workaround.
